src/Plugin/Field/FieldWidget/FileEntityBrowser.php

```diff
@@ -0,0 +1,254 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\file_entity\Plugin\Field\FieldWidget\FileEditableWidget.
+ */
+
+namespace Drupal\file_entity\Plugin\Field\FieldWidget;
+
+use Drupal\Component\Utility\SortArray;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+use Drupal\entity_browser\Plugin\Field\FieldWidget\EntityReference;
+
+/**
+ * Entity browser file widget.
+ *
+ * @FieldWidget(
+ *   id = "file_entity_browser",
+ *   label = @Translation("Entity browser (file)"),
+ *   provider = "entity_browser",
+ *   multiple_values = TRUE,
+ *   field_types = {
+ *     "file",
+ *     "image"
+ *   }
+ * )
+ */
+class FileEntityBrowser extends EntityReference {
+
+  /**
+   * Due to the table structure, this widget has a different depth.
+   *
+   * @var int
+   */
+  protected static $deleteDepth = 3;
+
+  /**
+   * @var \Drupal\Core\Field\FieldItemListInterface
+   */
+  protected $items;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultSettings() {
+    $settings = parent::defaultSettings();
+    $settings['view_mode'] = 'thumbnail';
+    return $settings;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+    $element = parent::settingsForm($form, $form_state);
+
+    $element['field_widget_display']['#access'] = FALSE;
+    $element['field_widget_display_settings']['#access'] = FALSE;
+
+    $element['view_mode'] = [
+      '#title' => t('File view mode'),
+      '#type' => 'select',
+      '#default_value' => $this->getSetting('view_mode'),
+      '#options' => \Drupal::service('entity_display.repository')->getViewModeOptions('file'),
+    ];
+
+    return $element;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsSummary() {
+    $summary = [];
+    $entity_browser_id = $this->getSetting('entity_browser');
+    $view_mode = $this->getSetting('view_mode');
+
+    if (empty($entity_browser_id)) {
+      return [t('No entity browser selected.')];
+    }
+    else {
+      $browser = $this->entityManager->getStorage('entity_browser')
+        ->load($entity_browser_id);
+      $summary[] = t('Entity browser: @browser', ['@browser' => $browser->label()]);
+    }
+
+    if (!empty($view_mode)) {
+      $summary[] = t('View mode: @name', ['@name' => $view_mode]);
+    }
+
+    return $summary;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
+    $this->items = $items;
```

Comment:
I thought I provided an update to add $items also to the new method but apparently I didn't. I can provide an update for that or we can keep it here, shouldn't really a problem and there could always be someone that needs something in there that's not provided.